resources/lib/gui.py

```diff
@@ -1040,6 +1105,11 @@ def onClick(self, controlID):
                 
             # If playlists have been enabled for widgets, add them too
             if self.widgetPlaylists:
+                # Wait for playlists to be loaded
+                while LIBRARY.playlists() != True:
+                    xbmc.sleep( 100 )
```

Comment:
are you sure that this will never hang here? (didnt look up that method)
Apart from that, all looks fine. :)